src/modules/partition/Config.h

```diff
@@ -162,6 +181,7 @@ public Q_SLOTS:
     SwapChoiceSet m_swapChoices;
     SwapChoice m_initialSwapChoice = NoSwap;
     SwapChoice m_swapChoice = NoSwap;
+    QString m_luksFileSystemType;
```

Comment:
I dunno, seems like an odd choice to have an enum, but store the string instead.

Reply:
I see your point. Originally I carried the `enum` all the way down to the `KPMHelpers.createNewEncryptedPartition` but that implied to include `Config.h` in a few places where I found less relevant, so I opted for a less complex storage but we are losing types I agree.

I see 3 options:
1. I rename `luksGeneration` into `luksFsType` with type: `QString` with possible values: `"luks" |  "luks2"` The enum doesn't bring a lot of value IMO.
2. I leave the enum be and carry it inside the `GlobalStorage` (will more complicated I think)
3. I leave the code as is

I would prefer 1 or 3 but could try 2 if you prefer.

Reply:
Let's definitely **not** do 2. GlobalStorage is not meant to store enums. It can do whatever QVariant supports, (so with suitable Qt futzing you can "make it fit"), but let's stick to ints, bools and strings there. That way we also don't need to think too hard about the interface from GS to Python. So 2 is out.

If it is most convenient to have a string you can pass in to KPMCore APIs (or other tooling), then by all means keep that.